src/recipe/recipe.dao.ts

```diff
@@ -16,6 +81,26 @@ export class RecipeDAO {
       },
     });
 
+    if (!recipeFromDb) {
+      throw new NotFoundException();
```

Comment:
you should not throw notFoundExceptions from Dao. Those should be used inside handlers